tidy3d/components/grid/grid_spec.py

```diff
@@ -351,6 +350,16 @@ def _make_coords_initial(  # pylint:disable=too-many-arguments,arguments-differ,
         # generate boundaries
         bound_coords = np.append(0.0, np.cumsum(np.concatenate(dl_list)))
         bound_coords += interval_coords[0]
+
+        # fix simulation domain boundaries which may be slightly off
+        domain_bounds = [bound[axis] for bound in symmetry_domain.bounds]
+        if not np.all(np.isclose(bound_coords[[0, -1]], domain_bounds)):
+            raise SetupError(
```

Comment:
I guess my only comments are:
* is there anything we can do to fix this if the condition is detected?
* if not, maybe we should add some advice to the user for things to try to get around the error (change resolution?)

Reply:
Basically it shouldn't happen. So I don't know what to advise... Apart from contact support?

Reply:
Like, some things may help but who knows. I guess we could advise switching to uniform grid and contacting us?

Reply:
ok, I guess we can just advise switching to uniform grid and contacting us. So basically it shouldn't happen given what we think now, but it's possible some weird edge case may trigger it that we don't know?

Reply:
Yeah, in which case I don't want to silently have a potentially wrong grid.

Reply:
How do you like the error message now?

Reply:
looks good